### DISCUSSION TITLE
Finding Java Thread Leaks With JDK Flight Recorder and a Bit Of SQL - Gunnar Morling

### DISCUSSION THREAD

**Discussion by giscus** [gunnarmorling/discussions.morling.dev#307] [Announcements]
# Finding Java Thread Leaks With JDK Flight Recorder and a Bit Of SQL - Gunnar Morling

The other day at work, we had a situation where we suspected a thread leak in one particular service,
i.e. code which continuously starts new threads, without taking care of ever stopping them again.
Each thread requires a bit of memory for its stack space,
so starting an unbounded number of threads can be considered as a form of memory leak, causing your application to run out of memory eventually.
In addition, the more threads there are, the more overhead the operating system incurs for scheduling them,
until the scheduler itself will consume most of the available CPU resources.
Thus it’s vital to detect and fix this kind of problem early on.

https://www.morling.dev/blog/finding-java-thread-leaks-with-jdk-flight-recorder-and-bit-of-sql/

**Comment by akash10agarwal** [comment]
The best such blog till date, can you pls do one for high CPU, high Mem, high GC as well?
